Mathlib/Topology/Algebra/FilterBasisNew.lean

```diff
@@ -0,0 +1,448 @@
+neighborhoods of the neutral element for some group (or ring, or module) topology.
+
+More precisely, for `X` in `{Group, AddGroup, Ring, Module}`, we provide :
+* `Filter.IsXBasis p s`: a strenghtening of `Filter.IsBasis` with compatibility conditions with the
+  algebraic structure.
+* `Filter.IsXBasis.mk_of_comm`: simpler contructors in the commutative setting.
+* `Filter.HasBasis.isXBasis`: in a topological `X`, *any* basis of neighborhoods
+  of the neutral element satisfies `Filter.IsXBasis`.
+* `Filter.IsXBasis.topologicalX_of_hasBasis`: if some `X` comes *equipped* with a topology and
+  a basis of neighborhoods of the neutral element satisfies `Filter.IsXBasis`, then that topology is
+  a `X` topology. This requires the topology to be invariant by translation.
+* `Filter.IsXBasis.topology`: the topological structure associated to some `X` filter basis.
+* `Filter.IsXBasis.topologicalX`: the topology associated to some `X` filter basis is indeed
+  an `X` topology.
+
+Note the subtlety between `IsXBasis.topologicalX_of_hasBasis`, useful for *proving* that some
+topology is compatible with the algebraic structure, and `IsXBasis.topologicalX` which shows
+that *the* topology associated to some basis is compatible with the algebraic structure.
+Mathematically this is the same thing, but having these variations is convenient.
+
+Remark: for `X = Module`, we use `continuousSMul` instead of `topologicalModule` in the names.
+
+## Usage indications
+
+The results in this file are meant to be used for either
+**proving some compatibility between topological and algebraic structures**
+or **constructing some topology compatible with an algebraic structure**.
+
+Thus, once you have all the relevant instances on your favorite type, you probably
+don't want to use this file anymore.
+
+## Implementation Notes
+
+For a long time, the definitions in this file were built on top of `FilterBasis`.
+We switched to `Filter.IsBasis` instead, because its API is much closer to
+that of `Filter.HasBasis`, which is what we use most of the time (when we already
+know the generated filter).
+
+## References
+
+* [N. Bourbaki, *General Topology*][bourbaki1966]
+-/
+
+open Filter Set TopologicalSpace Function
+
+open Topology Filter Pointwise
+
+universe u
+
+namespace Filter
+
+/-!
+## Filter bases for group topologies
+-/
+
+/-- Given a group `G`, `s : ι → Set G` and `p : ι → Prop`, `Filter.IsGroupBasis p s` is a
+strengthening of `Filter.IsBasis p s` with extra compatibility axioms ensuring that the generated
+filter is the filter of neighborhoods of one for some group topology on `G`
+(`Filter.IsGroupBasis.topologicalGroup`). Conversely, *any* basis of neighborhoods of one
+in a topological group satisfies these conditions (`Filter.HasBasis.isGroupBasis`). -/
+structure IsGroupBasis {G : Type*} {ι : Sort*} [Group G] (p : ι → Prop) (s : ι → Set G)
+    extends IsBasis p s : Prop where
+  one : ∀ {i}, p i → (1 : G) ∈ s i
```

Comment:
Can you please add docstrings to each field ? Maybe the linter will request that anyway…